src/coreclr/jit/codegenxarch.cpp

```diff
@@ -6232,6 +6232,11 @@ void CodeGen::genCompareInt(GenTree* treeNode)
     {
         JITDUMP("Not emitting compare due to flags being already set\n");
     }
+    else if (canReuseFlags && emit->AreFlagsSetForSignJumpOpt(op1->GetRegNum(), emitTypeSize(type), tree))
+    {
+        JITDUMP("Not emitting compare due to sign being already set, follow up instr will transform jump\n");
+        tree->gtFlags |= GTF_RELOP_SJUMP_OPT;
```

Comment:
Can you either always clear this new flag up above before the `if` on line 6231, or assert there that it's not set?

Reply:
Added the assertion next to the other ones. Please let me know if the comment needs to change or if its good enough.